plugins/modules/monitoring/newrelic_deployment.py

```diff
@@ -115,32 +120,57 @@ def main():
     if module.params["app_name"] and module.params["application_id"]:
         module.fail_json(msg="only one of 'app_name' or 'application_id' can be set")
 
+    app_id = None
     if module.params["app_name"]:
-        params["app_name"] = module.params["app_name"]
+        app_id = get_application_id(module)
     elif module.params["application_id"]:
-        params["application_id"] = module.params["application_id"]
+        app_id = module.params["application_id"]
     else:
         module.fail_json(msg="you must set one of 'app_name' or 'application_id'")
 
-    for item in ["changelog", "description", "revision", "user", "appname", "environment"]:
+    if app_id is None:
+        module.fail_json(msg="No application with name %s is found in NewRelic" % quote(module.params["app_name"], safe=''))
+
+    for item in ["changelog", "description", "revision", "user"]:
         if module.params[item]:
             params[item] = module.params[item]
 
     # If we're in check mode, just exit pretending like we succeeded
     if module.check_mode:
         module.exit_json(changed=True)
 
-    # Send the data to NewRelic
-    url = "https://rpm.newrelic.com/deployments.xml"
-    data = urlencode(params)
+    # Send the data to New Relic
+    url = "https://api.newrelic.com/v2/applications/%s/deployments.json" % quote(str(app_id), safe='')
+    data = {
+        'deployment': params
+    }
     headers = {
-        'x-api-key': module.params["token"],
+        'Api-Key': module.params["token"],
+        'Content-Type': 'application/json',
     }
-    response, info = fetch_url(module, url, data=data, headers=headers)
+    response, info = fetch_url(module, url, data=module.jsonify(data), headers=headers, method="POST")
     if info['status'] in (200, 201):
         module.exit_json(changed=True)
     else:
-        module.fail_json(msg="unable to update newrelic: %s" % info['msg'])
+        module.fail_json(msg="Unable to update New Relic: %s" % info['msg'])
+
+
+def get_application_id(module):
+    url = "https://api.newrelic.com/v2/applications.json"
+    data = "filter[name]=%s" % quote(module.params["app_name"], safe='')
+    headers = {
+        'Api-Key': module.params["token"],
+    }
+    response, info = fetch_url(module, url, data=data, headers=headers)
+    if info['status'] not in (200, 201):
+        module.fail_json(msg="Unable to get application from New Relic: %s" % info['msg'])
+        return None
+
+    result = json.loads(response.read())
+    if result is None or len(result["applications"]) == 0:
```

Comment:
Is it possible that the result is *not None* and also *does not contain the key "applications"*? If so, then we should 
```suggestion
    if result is None or len(result.get("applications", "")) == 0:
```